src/terminal_box.rs

```diff
@@ -406,72 +406,69 @@ where
                                 renderer.fill_quad(underline2_quad, line_color);
                             }
 
-                            if metadata.flags.contains(Flags::DOTTED_UNDERLINE) {
-                                let bottom_offset = style_line_height * 2.0;
-
+                            let mut draw_repeated = |rects: &[(f32, Option<f32>)]| {
```

Comment:
Maybe add a comment describing the meaning of the tuple fields, or use a struct, although maybe that's an overkill.